BrainPortal/lib/cbrain_task_generators/templates/portal.rb.erb

```diff
@@ -491,18 +495,26 @@ class CbrainTask::<%= name %> < <%= (descriptor['custom'] || {})['cbrain:inherit
 
   # Returns all the cbcsv files present (i.e. set by the user as inputs), as tuples (id, Userfile)
   def cbcsv_files
+    # This determines if the program expect to only read its input files,
+    # or modify them (:write). Depending on the value, more or less files
+    # are allowed to be processed.
+    access = self.class.properties[:readonly_input_files].present? ? :read : :write
     <%= "files = [%s]" % files.map { |f| ":'#{f['id']}'" }.join( ', ' ) %>
     <%= "file_lists = [%s]" % file_lists.map { |f| ":'#{f['id']}'" }.join( ', ' ) %>
     return [] if files.nil? || files.length == 0
     files.select { |f| ! self.params[f].nil? && ! file_lists.include?(f) } # Prevent problems with file-type inputs with list=true
-         .map    { |f| [f, Userfile.find_accessible_by_user(self.params[f], self.user)] }
+         .map    { |f| [f, Userfile.find_accessible_by_user(self.params[f], self.user, :access_requested => access)] }
          .select { |f| f[1].name.end_with?('.cbcsv') || f[1].suggested_file_type == CbrainFileList }
   end
 
   # Final set of tasks to be launched based on this task's parameters. Only
   # useful if the parameters set for this task represent a set of tasks
   # instead of just one.
   def final_task_list #:nodoc:
+    # This determines if the program expect to only read its input files,
```

Comment:
Maybe better to avoid the repetition. In case we change the logic in the future.

Reply:
Yes, I hate the fact that I had to repeat this several times. I can refactor into a method though, and the documentation would go int here. In fact, I'll do that.